controlpanel/frontend/views/tool.py

```diff
@@ -54,18 +56,20 @@ class DeployTool(LoginRequiredMixin, RedirectView):
     def get_redirect_url(self, *args, **kwargs):
         name = self.kwargs["name"]
 
-        try:
-            Tool.create(name).deploy_for(self.request.user)
+        async_to_sync(channel_layer.send)(
+            'background_tasks',
+            {
+                'type': 'tool.deploy',
```

Comment:
I imagine the type `tool.deploy` will somehow trigger `BackgroundTaskConsumer#tool_deploy()`, is this just a naming convention or we define this mapping somewhere? Didn't see anything in `control-panel/controlpanel/routing.py` - not a problem, I'm just curious.

Reply:
Oh yes, it's a [naming convention](https://channels.readthedocs.io/en/latest/topics/consumers.html#basic-layout) 👍